### PR TITLE
[PyCDE][NFC] Hide the `circt` package in our namespace

### PR DESCRIPTION
**Pull request by teqdruid**
Teaches PyCDE how to play nice in the python global namespace. CIRCT is now "hidden" as `pycde.circt`. Combined with #4546 (which integrates `mlir.*` into `circt`), PyCDE now exposes only a single package, `pycde`. This also fixes pycde after the same PR made a bunch of breaking changes.

### CODE REVIEW

File: frontends/PyCDE/src/CMakeLists.txt [hunk 1]
```diff
@@ -6,42 +6,79 @@
...
+  INSTALL_COMPONENT CIRCTBindingsPythonModules
+  INSTALL_DESTINATION python_packages/pycde/pycde/circt/_mlir_libs
+  OUTPUT_DIRECTORY "${PYCDE_PYTHON_PACKAGE_DIR}/pycde/circt/_mlir_libs"
+  RELATIVE_INSTALL_ROOT "../../.."
+  DECLARED_SOURCES
+    MLIRPythonSources.Core
+    CIRCTBindingsPythonExtension
+)
+
+add_mlir_python_modules(PyCDE_CIRCTPythonModules
```
Comment (mikeurbach) [inline frontends/PyCDE/src/CMakeLists.txt:62]: Does this need to be duplicated in PyCDE's CMakeLists? I would have been nice to reuse the original CIRCTPythonModules, maybe with some tweaks to the flags passed to add_mlir_python_modules. But I assume doing it this way was required.

**Review by mikeurbach — APPROVED** [review]
Doing this totally makes sense, so +1 on the current PR. I did have one question, but I don't know if that use-case is supported, so I assume you ended up implementing it this way for a good reason.

**Comment by teqdruid** [comment]
> I assume you ended up implementing it this way for a good reason.

Never assume that :p

File: frontends/PyCDE/src/CMakeLists.txt (hunk 1 above)
Comment (teqdruid) [reply frontends/PyCDE/src/CMakeLists.txt:62]: The current `CIRCTPythonModules` puts everything in `circt_core/circt`, not `pycde/pycde/circt`.

**Comment by teqdruid** [comment]
Thanks for the late Friday evening review!

**Pull request merged**